.github/workflows/ci_cd_release.yml

```diff
@@ -37,43 +34,35 @@ jobs:
         if [[ ${{ github.ref_name }} =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
           # Split the tag into its components
           IFS='.' read -ra PARTS <<< "${{ github.ref_name }}"
-          echo "MAJOR=${PARTS[0]}" >> $GITHUB_ENV
-          echo "MINOR=${PARTS[1]}" >> $GITHUB_ENV
-          echo "PATCH=${PARTS[2]}" >> $GITHUB_ENV
+          echo "X=${PARTS[0]}" >> $GITHUB_ENV
+          echo "Y=${PARTS[1]}" >> $GITHUB_ENV
+          echo "Z=${PARTS[2]}" >> $GITHUB_ENV
         else
           echo "Invalid tag format. Expected vX.Y.Z but got ${{ github.ref_name }}"
           exit 1
         fi
 
-    - name: "Check tag is valid for current branch"
+    - name: "Verify tag was created in the proper release branch"
       run: |
-        if [[ ${{ github.event.base_ref }} != "refs/heads/release/${{ env.MAJOR }}.${{ env.MINOR }}" ]]; then
+        # Remove leading "v" from env.X
+        X_NO_V="${X:1}"
+        # Check if the tag was created in the proper branch
+        if [[ ${{ github.event.base_ref }} != "refs/heads/release/${X_NO_V}.${{ env.Y }}" ]]; then
           echo "::error::Tag ${{ github.ref_name }} does not match branch version. wrong branch."
           exit 1
         fi
 
-    - name: "Remove MAJOR tag"
+    - name: "Remove old tags"
       run: |
-        if git tag -l | grep -q "^${{ env.MAJOR }}$"; then
-          git push --delete origin ${{ env.MAJOR }}
-        else
-          echo -e "Tag ${{ env.MAJOR }} does not exist, skipping deletion."
-        fi
-
-    - name: "Remove MAJOR.MINOR tag"
-      run: |
-        if git tag -l | grep -q "^${{ env.MAJOR }}\.${{ env.MINOR }}$"; then
-          git push --delete origin ${{ env.MAJOR }}.${{ env.MINOR }}
-        else
-          echo -e "Tag ${{ env.MAJOR }}.${{ env.MINOR }} does not exist, skipping deletion."
-        fi
+          git push --delete origin ${{ env.X }}
+          git push --delete origin ${{ env.X }}.${{ env.Y }}
```

Comment:
Shouldn't we avoid running these operations if not needed? If the tag doesn't exist beforehand this will fail, that's why the if conditions were defined.